xapian-applications/omega/query.cc

```diff
@@ -651,24 +652,28 @@ html_strip(const string &str)
     return res;
 }
 
-// FIXME split list into hash or map and use that rather than linear lookup?
-static int word_in_list(const string& word, const string& list)
-{
-    string::size_type split = 0, split2;
+static unordered_map<string, int> word_to_occurrence;
+
+static void build_word_map(const string& list) {
+    string word;
     int count = 0;
-    while ((split2 = list.find('\t', split)) != string::npos) {
-	if (word.size() == split2 - split) {
-	    if (memcmp(word.data(), list.data() + split, word.size()) == 0)
-		return count;
-	}
-	split = split2 + 1;
-	++count;
+    for (size_t i = 0; i <= list.length(); ++i) {
+	if (list[i] == '\t' || list[i] == '\0') {
+	    if (word_to_occurrence.find(word) == word_to_occurrence.end()) {
```

Comment:
Your intent here isn't to find it, just to check if it exists, so `.count()` makes more sense to me. Unless @ojwb has reasons for avoiding it, I think it will make for clearer code.